apidef/oas/server.go

```diff
@@ -60,6 +60,11 @@ func (s *Server) ExtractTo(api *apidef.APIDefinition) {
 	}
 	if s.GatewayTags != nil {
 		s.GatewayTags.ExtractTo(api)
+
+		// Omit gateway tags if empty or disabled
+		if !s.GatewayTags.Enabled || len(s.GatewayTags.Tags) == 0 {
```

Comment:
Please use ShouldOmit function.

Reply:
I just removed a case, because it doesn't work for this structure. I think it doesn't handle the case where the GatewayTags are allocated (Tags field). As we do a nil check, this is type-safe and doesn't use reflection.